source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,15 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern[pattern_len - 1] == '*' && uriPattern[pattern_len - 2] == '/') {
+    return uriPattern.compare(0, pattern_len - 1, uri, pattern_len - 1) == 0;
```

Comment:
the problem here is, if ```uriPattern="foo.bar/*"``` and ```uri="foo.bar/baz"```, ```uriPattern.compare(0, pattern_len - 1, uri)``` would give negative result `-3`. And this cannot be distinguished from the case as `uriPattern="foo.aar"` and `uri="foo.bar"`, for which `compare` gives result `-1.` So I'll have to use strncmp() or pointer walk but again it's not C++11 style.